### PR TITLE
Add filter

### PR DESCRIPTION
**Pull request by afragen**
Adds filter to allow for user adding additional domains to clear.

Usage:

``` php
add_filter( 'wp_rest_cache_clear_extra', function( $clear_api_calls ) {
    $extra = array(
        array( 'rest_domain' => 'https://bitbucket.org' ),
        array( 'rest_domain' => 'https://gitlab.com' ),
    );

    return array_merge( $clear_api_calls, $extra );
}, 10, 1 );
```

### CODE REVIEW

**Comment by scarstens** [comment]
oooo fancy!!

**Comment by afragen** [comment]
But now I have a better 💡

File: wp-rest-cache.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ public function __construct() {
 			$this->load_classes();
 
 			// initialize plugin during init
-			add_action( 'init', array( $this, 'init' ) );
+			add_action( 'init', array( $this, 'init' ), 5 );
```
Comment (afragen) [inline wp-rest-cache.php:64]: BTW, this one is important.

Comment (scarstens) [reply wp-rest-cache.php:64]: this commit was already on develop, PR closed no longer needed

**Pull request closed without merging**